check_patroni/types.py

```diff
@@ -1,19 +1,28 @@
 import logging
-import urllib3
 
 import attr
 import nagiosplugin
-from typing import Any, Callable, List
+import requests
+import urllib3
+from typing import Any, Callable, List, Optional, Tuple, Union
 
 _log = logging.getLogger("nagiosplugin")
 
 
+class APIError(requests.exceptions.RequestException):
+    """This exception is raised when the rest api couldn't
+    be reached and we got a http status code different from 200.
+    """
+
+    pass
+
+
 @attr.s(auto_attribs=True, frozen=True, slots=True)
 class ConnectionInfo:
     endpoints: List[str] = ["http://127.0.0.1:8008"]
-    cert_file: str = "./ssl/benoit-dalibo-cert.pem"
-    key_file: str = "./ssl/benoit-dalibo-key.pem"
-    ca_cert: str = "./ssl/CA-cert.pem"
+    cert_file: Optional[str] = None
+    key_file: Optional[str] = None
```

Comment:
I would suggest:
```suggestion
    cert: Optional[Union[str, Tuple[str,str]]] = None
```
so that you don't need to handle various cases below and just pass the `cert` field when it's not `None`.

Reply:
done in https://github.com/dalibo/check_patroni/pull/22/commits/568df1f00d2c332cce7497e297179f82bb64c558